File: demo/bridge-demo-adr/app/src/main/java/com/viethai/demo/MediaModuleBridge.kt
package com.viethai.demo

import android.webkit.JavascriptInterface
import android.webkit.WebView
import com.google.gson.Gson

/** Created by viethai.pham on 4/4/19 */
class MediaModuleBridge(webView: WebView, private val module: MediaModule, gson: Gson) :
  BaseModuleBridge(webView, gson) {

  @JavascriptInterface
  fun playVideo(requestString: String) {
    val request = this.gson.fromJson(requestString, Request::class.java)
    this.sendStreamResponse(request, this.module.playVideo())
  }
}

File: demo/bridge-demo-ios/BridgeDemoIOS/MediaModuleBridge.swift
//
//  MediaModuleBridge.swift
//  BridgeDemoIOS
//
//  Created by Viethai Pham on 5/4/19.
//  Copyright © 2019 Viet Hai Pham. All rights reserved.
//

import WebKit

public final class MediaModuleBridge: BaseModuleBridge {
  private let module: MediaModule
  
  public init(module: MediaModule, delegate: BridgeDelegate) {
    self.module = module
    super.init(delegate)
  }
}

extension MediaModuleBridge: WKScriptMessageHandler {
  public func userContentController(_ userContentController: WKUserContentController,
                                    didReceive message: WKScriptMessage) {
    let params = message.body as! [String : Any]
    let method = params["method"] as! String
    let callback = params["callback"] as! String
    
    switch method {
    case "playVideo":
      let videoStream = self.module.playVideo()
      
      self.sendStreamResult(stream: videoStream, callback: callback)
        .disposed(by: self.disposeBag)
      
    default:
      break
    }
  }
}
